Rank.java

```diff
@@ -0,0 +1,32 @@
+import java.util.Comparator;
+
+class RankComparator implements Comparator<Rank> {
+
+  public int compare(Rank r1, Rank r2) {
+    if (r1.equals(r2)) {
+      return 0;
+    } else if (r1.ordinal() < r2.ordinal()) {
+      return -1;
+    } else {
+      return 1;
+    }
+  }
+
+}
+
+public enum Rank {
```

Comment:
Can make this implement `Comparable`, so that the `Comparator` above isn't needed.

Reply:
Apparently enums already implement Comparable, with the ordinal ordering.